internal/states/state.go

```diff
@@ -330,7 +330,12 @@ func (s *State) ProviderAddrs() []addrs.AbsProviderConfig {
 	m := map[string]addrs.AbsProviderConfig{}
 	for _, ms := range s.Modules {
 		for _, rc := range ms.Resources {
-			m[rc.ProviderConfig.String()] = rc.ProviderConfig
+			//m[rc.ProviderConfig.String()] = rc.ProviderConfig // Ronny TODO: I assume we always have an instance inside the resource. Is it a valid assumption?
```

Comment:
For now, I've always seen we have the instances set in this part, but I'm keeping this comment here as a reminder to make sure this is correct in all scenarios (or if you guys can think of a scenario when this is not true).